source/adapters/level_zero/v2/event_provider_normal.cpp

```diff
@@ -0,0 +1,119 @@
+//===--------- event_provider_normal.cpp - Level Zero Adapter -------------===//
+//
+// Copyright (C) 2024 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include "event_provider_normal.hpp"
+#include "../common.hpp"
+#include "../context.hpp"
+#include "event_provider.hpp"
+#include "ur_api.h"
+#include "ze_api.h"
+#include <memory>
+
+namespace v2 {
+static constexpr int EVENTS_BURST = 64;
+
+provider_pool::provider_pool(ur_context_handle_t context,
+                             ur_device_handle_t device, event_type events,
+                             queue_type queue) {
+  ZeStruct<ze_event_pool_desc_t> desc;
+  desc.count = EVENTS_BURST;
+  desc.flags = 0;
+
+  if (events == event_type::EVENT_COUNTER) {
+    ze_event_pool_counter_based_exp_desc_t counterBasedExt = {
+        ZE_STRUCTURE_TYPE_COUNTER_BASED_EVENT_POOL_EXP_DESC, nullptr};
```

Comment:
Shouldn't the declaration be moved up to avoid storing a pointer to an out-of-scope object in `desc.pNext`?

Reply:
https://github.com/oneapi-src/unified-runtime/pull/1976